lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -59,7 +59,7 @@ def create_ptr_record(fqdn, ptr)
 
     def do_create(name, value, type)
       zone = get_zone(name)
-      unless create_record(zone['id'], name, type, value) and rectify_zone(zone['name'])
+      unless create_record(zone['id'], name, type, value) and update_soa(zone['id'], zone['name']) and rectify_zone(zone['name'])
```

Comment:
I suspect this will break the REST backend unless you implement a noop function for that. Maybe it's also time for a post-save helper function that does update_soa and rectify_zone?